### PR TITLE
fix(tui): provider color overrides across ranks + CLI / sessions / app.rs cleanup

### PR DESCRIPTION
**Pull request by junhoyeo**
## Summary

Follow-ups to #436 (cost-ranked shade palettes) plus adjacent cleanups surfaced during review.

### PR #436 bug fixes (`fix(tui)`)

- **[HIGH]** `[colors.providers]` override was silently disabled for rank ≥ 1. `get_provider_shade()` now consults `TokscaleConfig::get_provider_color()` at every rank and derives a 7-step lighten-to-white palette (`shade_from_base`) from the override base when present. A user setting `anthropic = \"#FF0000\"` now sees red-family shades across *all* their Anthropic models.
- **[MEDIUM]** Shade collisions under `GroupBy::WorkspaceModel`. Same model appearing in N workspaces used to consume N rank slots and then collide on insert. Now aggregated to `(provider, model_name) -> total_cost` before ranking.
- **[MEDIUM]** Non-deterministic order on cost ties caused shade flicker across refreshes. Secondary sort by model name makes assignment stable.
- **[LOW]** `CURSOR_SHADES` expanded from 1 → 7 entries; unknown-provider fallback replaced by a gray ramp (was collapsing every unknown-provider model to pure white).

13 new tests cover: rank-0-to-highest-cost, workspace dedupe, cost-tie determinism, NaN-cost handling, cross-provider independence, rebuild-on-update, plus `shade_from_base` edge cases and unknown/cursor palette regressions.

### CLI refactor (`refactor(cli)`)

19 client-filter bools + 6 date-range options were redeclared across 7 clap structs — ~145 duplicated `#[arg(...)]` lines plus 19-field `ClientFlags { ... }` rebuilds at every call site. The existing `ClientFlags` runtime helper becomes a `#[derive(Args)]` struct; a parallel `DateRangeFlags` captures the date block. Both are `#[command(flatten)]`ed into `Cli`, `Models`, `Monthly`, `Hourly`, `Graph`, `Tui`, and `Submit`.

**main.rs: 5093 → 4445 (-648 lines, -12.7%)**. No CLI surface change — all `--opencode`, `--claude`, `--today`, `--since`, etc. still parse identically. One intentional UX win: Submit's inconsistent \"Include only X data\" help text now matches the uniform \"Show only X usage\" used elsewhere.

### Sessions + TUI cleanup (`refactor(sessions)`, `refactor(tui)`)

An architect review rejected broader refactors (`SessionParser` trait, full app.rs split) as over-engineering, but flagged two narrow wins that landed:

- **11 session parsers** now use shared `open_readonly_sqlite` + `read_file_or_none` helpers in `sessions/utils.rs`. No behavior change — same read-only SQLite flags, same error-to-empty fallback.
- **`build_model_shade_map`** and **`build_export_json`** moved out of `App` into `tui/colors.rs` and `tui/export.rs` as pure functions. app.rs 2247 → 2174.

## Test plan

- [x] `cargo clippy --workspace --all-features --all-targets -- -D warnings` — clean
- [x] `cargo fmt --all -- --check` — clean
- [x] `cargo test --workspace --all-features` — 950 passing (345 + 83 + 519 + 3), no regressions
- [x] `tokscale models --help` spot-check: all client/date flags preserved
- [x] `tokscale submit --help`: unified \"Show only X usage\" help text
- [ ] Manual TUI verification with a `~/.tokscale` override for `anthropic` (would confirm HIGH bug fix end-to-end)

---

<a href="https://app.devin.ai/review/junhoyeo/tokscale/pull/440" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

---
## Summary by cubic
Fixes TUI shade palette bugs, makes model colors provider-aware across all views and reloads, and removes duplicated CLI/session code. No CLI surface changes; TUI colors now respect provider overrides, handle empty providers, and keep labels/colors when loading legacy caches.

- **Bug Fixes**
  - `[colors.providers]` overrides now apply at every rank via a 7-step palette (rank 0 = base).
  - Shade map is keyed and ranked by aggregated (provider, model) cost; ties sort by model name; NaN treated as 0. Duplicate model names across providers no longer collide.
  - Models, Overview, and Stats use provider-aware color lookup. Hourly data and cache retain `provider`/`displayName`/`colorKey` (backward-compatible) so colors persist across reloads.
  - Empty or merged provider IDs are normalized during shade-map build and lookup to prevent mismatches and rank-0 fallbacks.
  - Legacy hourly cache entries missing display fields now fall back to the map key to rebuild `displayName`/`colorKey`, preserving labels and colors.
  - `cursor` palette expanded to 7 shades; unknown providers use a neutral gray ramp. Tests cover ranking, dedupe, ties, NaN handling, provider separation, rebuilds, palette overrides, and legacy cache fallback.

- **Refactors**
  - CLI: Extracted `ClientFlags` and `DateRangeFlags` as `clap` `Args`, flattened into `Cli`, `Models`, `Monthly`, `Hourly`, `Graph`, `Tui`, `Submit`, and `Wrapped`. ~648 lines removed; flags unchanged; help text unified to “Show only X usage”.
  - Sessions (`@tokscale-core`): Added `open_readonly_sqlite` and `read_file_or_none`; applied across 11 parsers. No behavior change.
  - TUI (`@tokscale-cli`): Moved shade-map builder to `tui/colors.rs` and export JSON builder to `tui/export.rs` as pure functions.

<sup>Written for commit 4b2e2a3eccbdaedcb6ded7be29839c59a10284ae. Summary will update on new commits.</sup>

### CODE REVIEW

File: crates/tokscale-cli/src/tui/colors.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    map
+}
+
+fn provider_color_key<'a>(provider: &'a str, model: &'a str) -> &'a str {
+    if provider.contains(", ") {
+        get_provider_from_model(model)
+    } else {
+        provider
+    }
```
[the remark this answers is not in the document]
Comment (junhoyeo) [reply crates/tokscale-cli/src/tui/colors.rs:51]: Fixed in 03e6fff.

`provider_color_key()` now treats empty provider IDs the same way lookup normalization does (`provider.is_empty() || provider.contains(", ")`), so shade-map construction and `model_color_for()` produce the same `(provider, model)` key even when parsers emit an empty provider string.

File: crates/tokscale-cli/src/tui/colors.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                model_shade_key(provider, name),
+                get_provider_shade(provider, rank),
+            );
+        }
+    }
+    map
+}
+
+fn provider_color_key<'a>(provider: &'a str, model: &'a str) -> &'a str {
+    if provider.contains(", ") {
```
[the remark this answers is not in the document]
Comment (junhoyeo) [reply crates/tokscale-cli/src/tui/colors.rs]: Addressed in 03e6fff.

This now normalizes empty provider IDs during shade-map construction as well, which keeps the build-time key consistent with lookup-time normalization and preserves ranked shading for inferred providers.

File: crates/tokscale-cli/src/tui/cache.rs [hunk 3]
```diff
@@ -337,7 +346,24 @@ impl From<&HourlyModelInfo> for CachedHourlyModelInfo {
 
 impl From<CachedHourlyModelInfo> for HourlyModelInfo {
     fn from(h: CachedHourlyModelInfo) -> Self {
+        let display_name = if h.display_name.is_empty() {
```
[the remark this answers is not in the document]
Comment (junhoyeo) [reply crates/tokscale-cli/src/tui/cache.rs]: Addressed in cc73f8a.

The hourly cache path now mirrors the daily-cache pattern: deserialization gets the BTreeMap key and uses it as the fallback source for legacy `display_name` / `color_key` recovery, with a regression test to lock that behavior in.

File: crates/tokscale-cli/src/tui/cache.rs [hunk 4]
```diff
@@ -337,7 +346,24 @@ impl From<&HourlyModelInfo> for CachedHourlyModelInfo {
...
+            h.color_key
+        };
+
         Self {
+            provider: h.provider,
+            display_name,
+            color_key,
             tokens: h.tokens.into(),
             cost: h.cost,
         }
```
[the remark this answers is not in the document]
Comment (junhoyeo) [reply crates/tokscale-cli/src/tui/cache.rs]: Fixed in cc73f8a.

I replaced the direct `From<CachedHourlyModelInfo>` fallback with `hourly_model_info_from_cached(key, value)` so legacy hourly cache entries can recover `display_name` and `color_key` from the model map key. I also added a regression test covering an old hourly cache entry with the new metadata fields omitted.
